lib/include/private/aws_iot_defender_internal.h

```diff
@@ -272,6 +272,13 @@
     ( millisecondValue ) / 1000                \
 
 
+/**
+ * MQTT connection mock for unit testing defender API.
+ * This mock will be only used inside defender MQTT wrapper.
+ */
+
+#define DEFENDER_MQTT_CONNECTION_MOCK    ( ( IotMqttConnection_t ) 0xDEF30CED )
+
```

Comment:
Shouldn't this be wrapped with some sort of TEST flag?

Reply:
Yes, I didn't get time to see if there is an existing test flag.   I was also waiting to see if this concept is OK. It avoids unnecessary connects to brokers during testing.

Reply:
There is already a mocked MQTT implementation.  See if you can use that.